provider/provider.go

```diff
@@ -0,0 +1,41 @@
+/* provider is a Terraform provider for MAAS.
+ */
+package provider
+
+import (
+	"github.com/hashicorp/terraform/terraform"
```

Comment:
We will want to make sure we target this to `0.11.14`. For now we aren't supporting terraform `0.12`.